### DISCUSSION TITLE
How to read the docker file storage directly without docker daemon

### DISCUSSION THREAD

**Discussion by bsalunke** [moby/moby#44311] [Q&A] [answered]
I have a requirement where I need to read the docker file storage from the "/var/lib/docker/*" for all the docker images and containers without having "docker daemon" running on the host. 
it just like reading the image/container data from docker file system statically. 

Is there any spec or tool that support reading this file system? 

thanks!

**Comment by cpuguy83** [comment]
Not that I am aware of, and the format is not necessarily stable either.

**Comment by thaJeztah** [comment, marked as answer]
Files inside `/var/lib/docker` (or the location configured to use for this) are designed to be exclusively accessed by the `dockerd` daemon. While _possible_ to implement something to read those files through other means, it's not supported for that reason.
